### PR TITLE
Fix docker image

### PR DESCRIPTION
**Pull request by Luni-4**
This PR fixes the docker build task.

Thanks in advance for your review! :)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,19 +2,26 @@
...
-  cd /src/taskboot && \
-  mkdir -p ${BUILD_DIR} && \
-  pip install --no-cache-dir --build ${BUILD_DIR} . && \
-  rm -rf ${BUILD_DIR}
+RUN mkdir -p ${BUILD_DIR}
+
+# Define the working directory
+WORKDIR taskboot
+
+# Install taskboot from mounted source code
```
Comment (marco-c) [inline Dockerfile:19]: This is not mounted anymore, it's a real directory in the Docker container

**Comment by marco-c** [comment]
The main drawback of this change is that we'll have a bigger Docker image, since we copy the source to the Docker container, while previously we were not (as we were using the source from a volume shared with the host).
It should be a small difference though, so I'm OK with it.

File: Dockerfile [hunk 2]
```diff
@@ -2,19 +2,26 @@
...
+RUN mkdir -p ${MOUNT_SOURCE_CODE}
+COPY . ${MOUNT_SOURCE_CODE}
```
Comment (marco-c) [inline Dockerfile:21]: Just do `COPY . /src/taskboot` without the mkdir first, it should work anyway

File: Dockerfile [hunk 3]
```diff
@@ -2,19 +2,26 @@
...
 RUN apk add --no-cache git skopeo docker
 
-# Install taskboot from mounted source code
 # Use a dedicated build dir to avoid building in bind mount
-RUN --mount=type=bind,target=/src/taskboot \
-  cd /src/taskboot && \
-  mkdir -p ${BUILD_DIR} && \
-  pip install --no-cache-dir --build ${BUILD_DIR} . && \
-  rm -rf ${BUILD_DIR}
+RUN mkdir -p ${BUILD_DIR}
```
Comment (marco-c) [inline Dockerfile:14]: Run this mkdir directly in the last command, the one with pip, so at the end of the build this directory will not be in the image.

File: Dockerfile [hunk 4]
```diff
@@ -2,19 +2,26 @@
...
+
+# Define the working directory
+WORKDIR taskboot
+
+# Install taskboot from mounted source code
+RUN mkdir -p ${MOUNT_SOURCE_CODE}
+COPY . ${MOUNT_SOURCE_CODE}
+
+RUN cd ${MOUNT_SOURCE_CODE} && \
+    pip install --no-cache-dir --build ${BUILD_DIR} . && \
```
Comment (marco-c) [inline Dockerfile:24]: Nit: no need to cd first, you can just do `pip install --no-cache-dir --build ${BUILD_DIR} /src/taskboot`

**Comment by marco-c** [comment]
Nit: I'd remove BUILD_DIR and MOUNT_SOURCE_DIR argument and just replace them with their values, since we don't really need them as ARGs.

**Comment by Luni-4** [comment]
> The main drawback of this change is that we'll have a bigger Docker image, since we copy the source to the Docker container, while previously we were not (as we were using the source from a volume shared with the host).
> It should be a small difference though, so I'm OK with it.

You're perfectly right, but unfortunately it seems there are problems using `BuildKit` as you can see from [here](https://github.com/docker-library/python/issues/477). I tried to replace `--mount` with `-v`, but I couldn't compile the `Dockerfile` anymore.

**Pull request merged**